src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/QueryDocument.cs

```diff
@@ -38,7 +38,7 @@ NpgsqlTransaction transaction
                     connection,
                     transaction
                 ),
-                0
+                await _sqlAction.GetTotalDocumentsByKey(resourceName, connection, transaction)
```

Comment:
This is an expensive operation - full cross-partition table scan. It should only happen if total count is requested.